bob-backend/src/pearl/disk_controller.rs

```diff
@@ -135,10 +138,49 @@ impl DiskController {
         }
     }
 
+    fn collect_iostat(&self) -> (f64, f64) {
+        let mut iops = 0.;
+        let mut iowait = 0.;
+        let ios = std::process::Command::new("iostat").arg("-dx").output();
+        match ios {
+            Ok(output) => {
+                if let (true, Ok(out)) = 
+                    (output.status.success(), String::from_utf8(output.stdout)) {
+                    for i in out.lines() {
+                        let lsp: Vec<&str> = i.split_whitespace().collect();
+                        if lsp.len() > 0 && self.disk_dev_name == lsp[0] {
+                            if let (Ok(rs), Ok(ws)) = 
+                                (lsp[1].replace(',', ".").parse::<f64>(), 
+                                lsp[7].replace(',', ".").parse::<f64>()) {
+                                iops = rs + ws;
+                            }
+                            if let (Ok(rwait), Ok(wwait)) = 
+                                (lsp[5].replace(',', ".").parse::<f64>(), 
+                                lsp[11].replace(',', ".").parse::<f64>()) {
+                                iowait = rwait + wwait;
+                            }
+                        }
+                    }
+                }
```

Comment:
There should be debug message in case iostat fails. You can try to include `stderr` in message in this case.